crates/sui-core/src/overload_monitor.rs

```diff
@@ -204,6 +211,37 @@ fn check_overload_signals(
     (overload_status, load_shedding_percentage)
 }
 
+// Return true if we should reject the txn with `tx_digest`.
+fn should_reject_tx(
+    load_shedding_percentage: u32,
+    tx_digest: TransactionDigest,
+    minutes_since_epoch: u64,
+) -> bool {
+    let mut hasher = XxHash64::with_seed(minutes_since_epoch);
+    hasher.write(tx_digest.inner());
+    let value = hasher.finish();
+    value % 100 < load_shedding_percentage as u64
+}
+
+// Checks if we can accept the transaction with `tx_digest`.
+pub fn overload_monitor_accept_tx(
+    load_shedding_percentage: u32,
+    tx_digest: TransactionDigest,
+) -> SuiResult {
+    // Using the minutes_since_epoch as the hash seed to allow rejected transaction's
+    // retry to have a chance to go through in the future.
+    let minutes_since_epoch = SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .expect("Sui did not exist prior to 1970")
+        .as_secs()
+        / 60;
```

Comment:
no need to address this now, but i wonder if a minute is on the long side? the longer the retry interval the more we incentivize people to mine transactions